### ISSUE TITLE
Test coverage: attester slashings with duplicated attesting indices

### ISSUE THREAD

**Issue by michaelsproul** [ethereum/eth2.0-specs#1486]
While updating to v0.9.1 I noticed that the spec allows validator indices to be duplicated in `IndexedAttestation`s. In Lighthouse our function to check the sorting of the indices was also (incorrectly) checking for duplicates, which wasn't picked up by any of our tests or the spec tests. I think it would be good to have a test vector with an attester slashing containing duplicated indices, if that is indeed the intended behaviour. For the signature, it's sufficient to add the signatures of the duplicated validators to the aggregate multiple times. Concretely, this attester slashing passes validation (pardon the Rust):

```rust
let sig1 = Signature::new(&data1.tree_hash_root(), domain, &keypairs[0].sk);
let sig2 = Signature::new(&data2.tree_hash_root(), domain, &keypairs[0].sk);

let mut agg1 = AggregateSignature::new();
agg1.add(&sig1);
agg1.add(&sig1);

let mut agg2 = AggregateSignature::new();
agg2.add(&sig2);
agg2.add(&sig2);

let attestation_1 = IndexedAttestation {
    attesting_indices: vec![0, 0].into(),
    data: data1,
    signature: agg1,
};

let attestation_2 = IndexedAttestation {
    attesting_indices: vec![0, 0].into(),
    data: data2,
    signature: agg2,
};

let attester_slashing = AttesterSlashing {
    attestation_1,
    attestation_2,
};
```

**Comment by protolambda** [comment]
Hey, sorry for getting to this a little bit late, we have been prioritizing phase1 and more immediate phase0 work.

The duplicate indices case is interesting, and if valid as indexed attestation it may very well be slashable too, but a real attestation with bitfield does not have this same problem.

So given that it wouldn't be derived from a valid attestation, I think it's better to explicitly disallow duplicate indices in the index attestation / slashings, and not run into danger of this easily unaccounted for case on mainnet. The Python `sorted` is not enough to filter out duplicates, but I will propose a change to enforce it.
And I wonder how I missed this, as we do have tests for replacing and adding/removing indices :sweat: 

And although a slight phase0 change, it shouldn't be breaking any client from interop unless they produce slashings with duplicate indices already, so I think we can go ahead and get it into the `v09x` branch.
Thanks @michaelsproul !

**Comment by michaelsproul** [comment]
That sounds like a good resolution to me!

**Comment by djrtwo** [comment]
addressed via #1516
